homeassistant/components/recorder/migration.py

```diff
@@ -2367,17 +2367,22 @@ def migrate_data(instance: Recorder) -> bool:
         _LOGGER.debug("Migrating entity_ids done=%s", is_done)
         return is_done
 
-    def migration_done(self, instance: Recorder) -> None:
+    def migration_done(self, instance: Recorder, _session: Session | None) -> None:
         """Will be called after migrate returns True."""
         # The migration has finished, now we start the post migration
         # to remove the old entity_id data from the states table
         # at this point we can also start using the StatesMeta table
         # so we set active to True
         _LOGGER.debug("Activating states_meta manager as all data is migrated")
         instance.states_meta_manager.active = True
+        session_generator = (
+            contextlib.nullcontext(_session)
+            if _session
+            else session_scope(session=instance.get_session())
+        )
```

Comment:
Can we always require a session here to avoid this branching?

Reply:
Maybe, but then we would always need to provide a session object to `migration_done`, which would mostly not be used.